src/sagemaker_xgboost_container/algorithm_mode/hyperparameter_validation.py

```diff
@@ -94,10 +94,9 @@ def eval_metric_range_validator(SUPPORTED_METRIC, metric):
     @hpv.dependencies_validator(["objective"])
     def eval_metric_dep_validator(value, dependencies):
         if "auc" in value:
-            if (dependencies["objective"] not in
-                ["binary:logistic", "binary:logitraw", "multi:softmax", "multi:softprob",
-                 "reg:logistic", "rank:pairwise", "binary:hinge"]):
-                raise exc.UserError("Metric 'auc' can only be applied for classification and ranking problem.")
+            if not any(metric_type in dependencies["objective"] for metric_type in [
```

Comment:
We might want to be little more specific by saying `dependencies["objective"].startswith(metric_type)`. It may not affect here since the objective will always start with `text:` but might be easier to understand and maintain.

Reply:
This is fair. I will change to starts with and squash merge.